### PR TITLE
add FilterList and Product Card on the Products page

### PR DESCRIPTION
**Pull request by hsnice16**
### This PR will add code for FilterList and Product Card on the Products page

In this PR, **515 additions and 76 deletions** lines of code.

- some files contain only 5 to 6 lines of code only
- you can avoid reviewing CSS files, backend related files

### Please review this PR, and give your feedback

[Live URL](https://react-inclusive-mart-puvdvjh8r-hsnice16.vercel.app/products)

image

### CODE REVIEW

File: src/components/products-filter-list/ClearAllButton.jsx [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  </button>
+);
+
+ClearAllButton.propTypes = {
+  forScreenSize: PropTypes.string,
+};
+
+ClearAllButton.defaultProps = {
+  forScreenSize: "",
+};
```
Comment (anik31) [inline src/components/products-filter-list/ClearAllButton.jsx:19]: Appreciation - good thing that you have used data type & default value for props

File: src/utils/constants/apis.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+export const API_TO_GET_ALL_PRODUCTS = {
+  api: "/api/products",
+  propertyToGet: "products",
+};
+
+export const API_TO_GET_HOME_PRODUCTS = {
+  api: "/api/products/home",
+  propertyToGet: "products",
+};
```
Comment (anik31) [inline src/utils/constants/apis.js:14]: Appreciation - this is very nice, you re maintaining an object for API calls & based on that calling a single hook. Even I will be using this.

File: src/custom-hooks/useAsync.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  const [state, dispatch] = useReducer(
+    sharedReducer,
+    sharedInitialReducerState
+  );
+
+  useEffect(() => {
+    (async () => {
+      dispatch({ type: ACTION_TYPE_LOADING });
+
+      try {
```
Comment (anik31) [inline src/custom-hooks/useAsync.js:30]: Appreciation - great, you have created a hook for calling APIs

**Review by anik31 — COMMENTED** [review]
Good work 💯 The code quality is very good since you have broken down everything into separate files. 

Suggestion - the price range is not clickable, please check that. Also, I tried to add filters in the live preview but it is not working.

**Comment by hsnice16** [comment]
> Good work 💯 The code quality is very good since you have broken down everything into separate files.
> 
> Suggestion - the price range is not clickable, please check that. Also, I tried to add filters in the live preview but it is not working.

Yeah, I am adding functionality to them. This PR was only for the design code.

Thanks for reviewing this.

**Review by BalajiHarinath — APPROVED** [review]
Appreciation : I liked the way you used a custom hook to fetch the data from the backend.

**Comment by hsnice16** [comment]
Thank you for reviewing this.

**Pull request merged**